src/cpp/rtps/transport/shared_mem/SharedMemManager.hpp

```diff
@@ -718,7 +718,9 @@ class SharedMemManager :
                         throw std::runtime_error("");
                     }
 
+                    // Read and pop descriptor
                     SharedMemGlobal::BufferDescriptor buffer_descriptor = head_cell->data();
+                    global_port_->pop(*global_listener_, was_cell_freed);
```

Comment:
taking out the message before failing to find the segment, and then continue to avoid receiving null message forever. ltgm.